src/library/scala/runtime/ScalaRunTime.scala

```diff
@@ -155,8 +155,23 @@ object ScalaRunTime {
   // More background at ticket #2318.
   def ensureAccessible(m: JMethod): JMethod = scala.reflect.ensureAccessible(m)
 
-  def _toString(x: Product): String =
-    x.productIterator.mkString(x.productPrefix + "(", ",", ")")
+  //x.productElementNames.zip(x.productIterator).map { case (n, v) => s"$n = $v" }.mkString(x.productPrefix + "(", ", ", ")")
+  def _toString0(x: Product): String = {
+    val sb = new java.lang.StringBuilder
+    val nm = x.productElementNames
+    val it = x.productIterator
+    sb.append(x.productPrefix).append("(")
+    var more = it.hasNext
+    while (more) {
+      sb.append(nm.next()).append(" = ").append(java.util.Objects.toString(it.next()))
+      more = it.hasNext
+      if (more) sb.append(", ")
```

Comment:
I recommend to avoid the spacers, either on the sides of `=` or after commas, or both. So just `Foo(a=true,b=-1)` rather than `Foo(a = true, b = -1)`. Can actually make a big difference for larger case classes

Reply:
Thanks for the feedback. It got a heart from dwijnand, who has a big heart. This style dilemma is the camel-back-breaking issue. I guessed that people like to write the apply with spaces.

Footnote: maybe compiler can infer from the source text how the user prefers to write it! Maybe for more than N params, output as ascii table!

Reply:
This is highly bike-sheddable. I like the spaces, since that's how it would appear in nearly every Scala codebase I've seen.

Reply:
That's two big hearts for.